cmd/net-crd-installer/utils/util.go

```diff
@@ -0,0 +1,158 @@
+	// FleetLabelValue is the value for the AzureManagedLabelKey indicating management by Fleet.
+	FleetLabelValue = "fleet"
+)
+
+var (
+	// memberIncludedCRDs defines CRDs that should be included in member clusters
+	memberIncludedCRDs = map[string]bool{
+		"serviceexports.networking.fleet.azure.com":       true,
+		"serviceimports.networking.fleet.azure.com":       true,
+		"multiclusterservices.networking.fleet.azure.com": true,
+	}
+)
+
+// InstallCRD creates/updates a Custom Resource Definition (CRD) from the provided CRD object.
+func InstallCRD(ctx context.Context, client client.Client, crd *apiextensionsv1.CustomResourceDefinition, isE2ETest bool) error {
+	klog.V(2).Infof("Installing CRD: %s", crd.Name)
+
+	existingCRD := apiextensionsv1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crd.Name,
+		},
+	}
+
+	createOrUpdateRes, err := controllerutil.CreateOrUpdate(ctx, client, &existingCRD, func() error {
+		// Copy spec from our decoded CRD to the object we're creating/updating.
+		existingCRD.Spec = crd.Spec
+
+		// Add an additional ownership label to indicate this CRD is managed by the installer.
+		if existingCRD.Labels == nil {
+			existingCRD.Labels = make(map[string]string)
+		}
+		// Ensure the label for management by the installer is set.
+		existingCRD.Labels[CRDInstallerLabelKey] = "true"
+		// Only set the Azure managed label if this is not an E2E test.
+		// This label is needed for clean up of CRD by kube-addon-manager in production,
+		// but can interfere with E2E test as CRDs are removed from AKS clusters which
+		// are actually not hub/member clusters.
+		if !isE2ETest {
+			existingCRD.Labels[AzureManagedLabelKey] = FleetLabelValue
+		}
+		return nil
+	})
+
+	if err != nil {
+		klog.ErrorS(err, "Failed to create or update CRD", "name", crd.Name, "operation", createOrUpdateRes)
+		return err
+	}
+
+	klog.Infof("Successfully created/updated CRD %s", crd.Name)
+	return nil
+}
+
+// CollectCRDs collects CRDs from the specified path based on the mode either hub/member.
+func CollectCRDs(crdDirectoryPath, mode string, scheme *runtime.Scheme) ([]apiextensionsv1.CustomResourceDefinition, error) {
+	// Set of CRDs to install based on mode.
+	crdsToInstall := []apiextensionsv1.CustomResourceDefinition{}
+
+	// Walk through the CRD directory and collect filenames.
+	if err := filepath.WalkDir(crdDirectoryPath, func(crdpath string, d fs.DirEntry, err error) error {
+		// Handle errors from WalkDir.
+		if err != nil {
+			return err
+		}
+
+		// Skip root directory i.e. config/crd/bases and any other directory since none should exist.
+		if d.IsDir() {
+			return nil
+		}
+
+		crd, err := GetCRDFromPath(crdpath, scheme)
+		if err != nil {
+			return err
+		}
+
+		group := crd.Spec.Group
+		// Check if the group equals "networking.fleet.azure.com".
+		if group == "networking.fleet.azure.com" {
+			crdName := crd.Name
+
+			if mode == "hub" {
+				crdsToInstall = append(crdsToInstall, *crd)
```

Comment:
Hi Arvind! Sorry if there's any misunderstanding, but do we need to install all APIs (e.g., ServiceExport, MCS) to the hub cluster as well?